src/boot/bootctl.c

```diff
@@ -1132,6 +1181,18 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_graceful = true;
                         break;
 
+                case ARG_MAKE_MID:
+                        if (!optarg || !strncmp(optarg, "yes", strlen("yes")))
+                                arg_make_mid = MID_YES;
+                        else if (!strncmp(optarg, "no", strlen("no")))
+                                arg_make_mid = MID_NO;
+                        else if (!strncmp(optarg, "auto", strlen("auto")))
+                                arg_make_mid = MID_AUTO;  /* default */
```

Comment:
please use parse_boolean() for parsing the yes/no part. i.e.:

```c
if (streq(optarg, "auto")) 
        arg_make_machine_id_directory = -1;
else {
        r = parse_boolean(optarg);
        if (r < 0)
                 …
        arg_make_machine_id_directory = r;
}
```

parse_boolean() is what we use everywhere else, and it allos specifying 1/0 as alternative syntaxes, too, and similar

Reply:
Ah yeah, moved to be pretty much this (with additional `!optarg` branch at the front since this is an `optional_argument`).